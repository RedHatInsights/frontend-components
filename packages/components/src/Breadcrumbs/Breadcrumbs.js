/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';

/**
 * @deprecated
 *
 * Breadcrumbs from FE component shouldn't be used anymore.
 *
 * Use <a href="https://www.patternfly.org/v4/components/breadcrumb" target="_blank">Breadcrumbs</a> from PF repository.
 */
const Breadcrumbs = ({ items, current, className, onNavigate, ...props }) => {
  console.warn(
    "Breadcrumbs from FE component shouldn't be used anymore. \
Instead use https://patternfly-react.surge.sh/documentation/react/components/breadcrumb from PF repository."
  );
  return (
    <Breadcrumb className={classnames('ins-c-breadcrumbs', className)} {...props}>
      {items.map((oneLink, key) => (
        <BreadcrumbItem key={key} data-key={key}>
          <a onClick={(event) => onNavigate(event, oneLink.navigate, key)} aria-label={oneLink.navigate}>
            {oneLink.title}
          </a>
        </BreadcrumbItem>
      ))}
      {current && <BreadcrumbItem isActive> {current} </BreadcrumbItem>}
    </Breadcrumb>
  );
};

Breadcrumbs.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      navigate: PropTypes.any,
      title: PropTypes.node,
    })
  ),
  current: PropTypes.node,
  onNavigate: PropTypes.func,
};

Breadcrumbs.defaultProps = {
  items: [],
  current: null,
  onNavigate: Function.prototype,
  className: '',
};

export default Breadcrumbs;
