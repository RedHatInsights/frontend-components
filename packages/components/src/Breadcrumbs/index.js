export { default } from './Breadcrumbs';
export { default as Breadcrumbs } from './Breadcrumbs';
export { default as ConnectedBreadcrumbs } from './ConnectedBreadcrumbs';
