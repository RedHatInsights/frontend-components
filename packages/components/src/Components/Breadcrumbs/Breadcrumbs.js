import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';

const Breadcrumbs = ({ items, current, className, onNavigate, ...props }) => {
    console.warn('Breadcrumbs from FE component shouldn\'t be used anymore. \
Instead use http://patternfly-react.surge.sh/patternfly-4/components/breadcrumb#Breadcrumb from PF repository.');
    return (
        <Breadcrumb className={ classnames('ins-c-breadcrumbs', className) } { ...props }>
            {
                items.map((oneLink, key) => (
                    <BreadcrumbItem key={ key } data-key={ key }>
                        <a onClick={ event => onNavigate(event, oneLink.navigate, key) }
                            aria-label={ oneLink.navigate }>
                            { oneLink.title }
                        </a>
                    </BreadcrumbItem>
                )
                ) }
            { current && <BreadcrumbItem isActive> { current } </BreadcrumbItem> }
        </Breadcrumb>
    );
};

Breadcrumbs.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        navigate: PropTypes.any,
        title: PropTypes.node
    })),
    current: PropTypes.node,
    onNavigate: PropTypes.func
};

Breadcrumbs.defaultProps = {
    items: [],
    current: null,
    onNavigate: Function.prototype,
    className: ''
};

export default Breadcrumbs;
