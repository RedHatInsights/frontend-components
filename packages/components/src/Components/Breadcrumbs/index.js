export { default as Breadcrumbs } from './Breadcrumbs';
export { default as ConnectedBreadcrumbs } from './ConnectedBreadcrumbs';
