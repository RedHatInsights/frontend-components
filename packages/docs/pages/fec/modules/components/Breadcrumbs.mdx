import DeprecationWarn from '@docs/deprecation-warn'

<DeprecationWarn />

# Breadcrumbs


Breadcrumbs from FE component shouldn't be used anymore.

Use <a href="https://www.patternfly.org/v4/components/breadcrumb" target="_blank">Breadcrumbs</a> from PF repository.

## Props

|name|type|default|description|
|----|----|-------|-----------|
|items|BreadcrumbItem[]|`[]`||
|current|React.ReactNode|`null`||
|onNavigate|(event:React.MouseEvent&lt;HTMLAnchorElement&gt;,link:string,key:number)=&gt;void|`() => undefined`||
|className|<code>undefined</code>|`''`||



