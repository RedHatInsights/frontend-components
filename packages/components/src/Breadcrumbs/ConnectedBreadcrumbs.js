/* eslint-disable react/prop-types */
import React from 'react';
import { useEffect } from 'react';
import { useHistory, useLocation, useRouteMatch } from 'react-router-dom';
import Breadcrumbs from './Breadcrumbs';

/**
 * @deprecated
 *
 * Breadcrumbs from FE component shouldn't be used anymore.
 *
 * Use <a href="https://www.patternfly.org/v4/components/breadcrumb" target="_blank">Breadcrumbs</a> from PF repository.
 */
const ConnectedBreadcrumbs = (props) => {
    const history = useHistory();
    const location = useLocation();
    const match = useRouteMatch();
    useEffect(() => {
        console.warn('This component will be removed in future release (next month March), do not use it anymore!');
    }, []);

    const onNavigate = (_event, _item, key) => {
        history.go(-key);
    };

    const calculateBreadcrumbs = () => {
        const { current, mappings } = props;
        if (!current && mappings) {
            const root = match.path.split('/').slice(2);
            const rest = location.pathname.substring(match.path.length).split('/').slice(1);
            return [
                ...root,
                ...rest.map((item, key) => mappings[key] || item)
            ];
        } else {
            return [
                ...location.pathname.split('/').slice(2, -1),
                current
            ];
        }
    };

    const mappedBreadcrumbs = calculateBreadcrumbs() || [];
    return (
        <Breadcrumbs { ...props }
            items={ mappedBreadcrumbs.slice(0, -1).map(item => ({ title: item, navigate: item })) }
            onNavigate={ onNavigate }
            current={ mappedBreadcrumbs.slice(-1)[0] }
        />
    );
};

export default ConnectedBreadcrumbs;
