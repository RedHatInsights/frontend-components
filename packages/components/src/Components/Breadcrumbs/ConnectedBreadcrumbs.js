import React, { Component } from 'react';
import Breadcrumbs from './Breadcrumbs';
import { withRouter } from 'react-router-dom';

class ConnectedBreadcrumbs extends Component {
    constructor(props) {
        super(props);
        this.onNavigate = this.onNavigate.bind(this);
        this.calculateBreadcrumbs = this.calculateBreadcrumbs.bind(this);
        console.warn('This component will be removed in future release (next month March), do not use it anymore!');
    }

    onNavigate(_event, _item, key) {
        const { history } = this.props;
        history.go(-key);
    }

    calculateBreadcrumbs() {
        const { match, location, current, mappings } = this.props;
        if (!current && mappings) {
            const root = match.path.split('/').slice(2);
            const rest = location.pathname.substring(match.path.length).split('/').slice(1);
            return [
                ...root,
                ...rest.map((item, key) => mappings[key] || item)
            ];
        } else {
            return [
                ...location.pathname.split('/').slice(2, -1),
                current
            ];
        }
    }

    render() {
        const { match, location, history, current, staticContext, dispatch, ...props } = this.props;
        const mappedBreadcrumbs = this.calculateBreadcrumbs() || [];
        return (
            <Breadcrumbs { ...props }
                items={ mappedBreadcrumbs.slice(0, -1).map(item => ({ title: item, navigate: item })) }
                onNavigate={ this.onNavigate }
                current={ mappedBreadcrumbs.slice(-1)[0] }
            />
        );
    }
}

export default withRouter(ConnectedBreadcrumbs);
