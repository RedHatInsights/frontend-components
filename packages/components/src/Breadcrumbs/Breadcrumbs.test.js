import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs component', () => {
    const onNavigate = jest.fn();
    const items = [
        { title: 'First', navigate: 'first' },
        { title: <span>Second</span>, navigate: 'second' }
    ];
    const current = 'Something';
    describe('should render correctly', () => {
        it('no items', () => {
            const wrapper = shallow(<Breadcrumbs items={undefined} current={current} onNavigate={onNavigate} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('no current', () => {
            const wrapper = shallow(<Breadcrumbs items={items} current={undefined} onNavigate={onNavigate} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('with items and current item', () => {
            const wrapper = shallow(<Breadcrumbs items={items} current={current} onNavigate={onNavigate} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('API', () => {
        it('should call correctly', () => {
            const wrapper = shallow(<Breadcrumbs items={items} current={current} onNavigate={onNavigate} />);
            wrapper.find('a').first().simulate('click');
            expect(onNavigate.mock.calls.length).toBe(1);
            expect(onNavigate.mock.calls[0][1]).toBe(items[0].navigate);
            expect(onNavigate.mock.calls[0][2]).toBe(0);
        });
    });
});
