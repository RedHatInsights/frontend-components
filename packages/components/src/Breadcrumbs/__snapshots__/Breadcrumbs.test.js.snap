// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should render correctly no current 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-v6-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-2"
    data-ouia-component-type="PF6/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-v6-c-breadcrumb__list"
      role="list"
    >
      <li
        class="pf-v6-c-breadcrumb__item"
        data-key="0"
      >
        <a
          aria-label="first"
        >
          First
        </a>
      </li>
      <li
        class="pf-v6-c-breadcrumb__item"
        data-key="1"
      >
        <span
          class="pf-v6-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <a
          aria-label="second"
        >
          <span>
            Second
          </span>
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumbs component should render correctly no items 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-v6-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
    data-ouia-component-type="PF6/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-v6-c-breadcrumb__list"
      role="list"
    >
      <li
        class="pf-v6-c-breadcrumb__item"
      >
         
        Something
         
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumbs component should render correctly with items and current item 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="pf-v6-c-breadcrumb ins-c-breadcrumbs"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-3"
    data-ouia-component-type="PF6/Breadcrumb"
    data-ouia-safe="true"
  >
    <ol
      class="pf-v6-c-breadcrumb__list"
      role="list"
    >
      <li
        class="pf-v6-c-breadcrumb__item"
        data-key="0"
      >
        <a
          aria-label="first"
        >
          First
        </a>
      </li>
      <li
        class="pf-v6-c-breadcrumb__item"
        data-key="1"
      >
        <span
          class="pf-v6-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <a
          aria-label="second"
        >
          <span>
            Second
          </span>
        </a>
      </li>
      <li
        class="pf-v6-c-breadcrumb__item"
      >
        <span
          class="pf-v6-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
         
        Something
         
      </li>
    </ol>
  </nav>
</div>
`;
